### DISCUSSION TITLE
Will running apps be forced closed?

### DISCUSSION THREAD

**Discussion by romnation1** [Romanitho/Winget-AutoUpdate#653] [General]
One thing i noticed is manually triggering winget to update a package via powershell force closes the app. Is this the same experience my users are going to experience when the daily check for 3rd party app updates runs? I do NOT want it to force close any running apps, especially if this is something we are going to have kick off daily or at logon, as we do not want to interrupt the user. Right now we have all notifications on silent as well.

**Comment by AndrewDemski-ad-gmail-com** [comment]
WAU triggers winget upgrade of application packages. Those are executing setups/installers created by manufacturers/vendors of that software.
If package is being distributed in non-interactive/silent mode, often logic developed by software authors kills processes without asking anyone who may or may not use it at the time o the upgrade.

You would need to prepare custom pre- & post- upgrade/installation mod scripts to pause upgrade, notify user that something is going to be installed/modified. 

WAU itself runs in system account context and requires workaround to that limitation using designated task in scheduler.

You may need to write something similar for each package which must be interactively closed by user before system-context upgrade/installation may proceed.
